src/sparseml/core/recipe/container.py

```diff
@@ -38,16 +38,89 @@
+
+    :param compiled_recipe: the compiled recipe from the recipes list
+    :param recipes: the list of RecipeTuple instances to be compiled
+    """
+
     compiled_recipe: Optional[Recipe] = None
     recipes: List[RecipeTuple] = field(default_factory=list)
 
     def update(
         self,
-        recipe: Union[str, List[str], Recipe, List[Recipe]] = None,
-        recipe_stage: Union[str, List[str], List[List[str]]] = None,
-        recipe_args: Union[Dict[str, Any], List[Dict[str, Any]]] = None,
+        recipe: Union[str, List[str], Recipe, List[Recipe], None] = None,
+        recipe_stage: Union[str, List[str], List[List[str]], None] = None,
+        recipe_args: Union[Dict[str, Any], List[Dict[str, Any]], None] = None,
         **kwargs,
     ) -> Dict:
+        """
+        Update the recipes in the container. If a recipe is provided, it will
+        reset any existing compiled_recipe in the container. Must call
+        `check_compile_recipe` to re-compile the recipes into a single compiled_recipe.
+        If no recipe is provided, does nothing and returns the kwargs.
+
+
+        No recipe provided:
+        >>> container = RecipeContainer()
+        >>> kwargs = dict(irrelevant_kwarg="prune")
+        >>> result = container.update(**kwargs)
+        >>> result == kwargs
+        True
+
+        With Recipe provided compiled_recipe is reset:
+        >>> from sparseml.core import Recipe
+        >>> recipe_str = '''
+        ... test_stage:
+        ...     pruning_modifiers:
+        ...         ConstantPruningModifier:
+        ...             start: 0.0
+        ...             end: 2.0
+        ...             targets: ['re:.*weight']
+        ... '''
+        >>> container = RecipeContainer(
+        ...    compiled_recipe=Recipe.create_instance(recipe_str)
+        ... )
+        >>> container.compiled_recipe is not None
+        True
+        >>> result = container.update(recipe=recipe_str)
+        >>> container.compiled_recipe is None
+        True
+
+        Recompile the updated container with check_compile_recipe:
+        >>> _ = container.check_compile_recipe()
+        >>> container.compiled_recipe is not None
+        True
+
+
+        Can provide multiple recipes to update the container with:
+        >>> container = RecipeContainer()
+        >>> recipe_str_1 = '''
+        ... test_stage:
+        ...     pruning_modifiers:
+        ...         ConstantPruningModifier:
+        ...             start: 0.0
+        ...             end: 2.0
+        ...             targets: ['re:.*weight']
+        ... '''
+        >>> recipe_str_2 = '''
+        ... test_stage:
+        ...     pruning_modifiers:
+        ...         ConstantPruningModifier:
+        ...             start: 3.0
+        ...             end: 4.0
+        ...             targets: ['re:.*weight']
+        ... '''
+        >>> result = container.update(recipe=[recipe_str_1, recipe_str_2])
+        >>> len(container.recipes)
+        2
```

Comment:
I think having all these tests in the documentation is a bit much, there were some other classes I noticed had this problem too. Could we limit the tests to one example and move the rest to unit tests?

Reply:
Done!